diffpy/snmf/stretchednmfapp.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import argparse
+
+from diffpy.snmf.io import load_input_signals, initialize_variables
+from diffpy.snmf.subroutines import update_weights_matrix, get_residual_matrix, objective_function
+
+
+def create_parser():
+    parser = argparse.ArgumentParser(
+        prog="stretched_nmf",
+        description="Stretched Nonnegative Matrix Factorization"
+    )
+    parser.add_argument('-v', '--version', action='version', help='Print the software version number')
+    parser.add_argument('-i', '--input-directory', type=str, default=None,
+                        help="Directory containing experimental data. Default before will cause the program to use the current working directory as the input directory.")
+    parser.add_argument('-o', '--output-directory', type=str,
+                        help="The directory where the results will be dumped. Default behavior will create a new directory named 'smnf_results' inside the input directory.")
+    parser.add_argument('-t', '--data-type', type=str, choices=['xrd', 'pdf'],
+                        help="The type of the experimental data.")
+    parser.add_argument('components', type=int,
+                        help="The number of component signals to obtain from experimental "
+                             "data. Must be an integer greater than 0.")
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = create_parser()
+
+    grid, data_input, data_type = load_input_signals(args.input_directory)
+    if args.data_type is not None:
```

Comment:
for readability I would separate the action of handling the default behavior and calling the function.  These are just general comments to make your code more readable, I hope you are taking them on board and reusing later...

What I mean is:
```
    grid, data_input, data_type = load_input_signals(args.input_directory)
    if args.data_type:
       data_type = args.data_type
   variables = initialize_variables(data_input, args.components, data_type)
```
if not args.data_type:
    try:

Reply:
I see what you mean. For example, it's not good practice to have the function call inside the if statement if that's not needed.